specs/altair/sync-protocol.md

```diff
@@ -153,43 +189,66 @@ def validate_light_client_update(snapshot: LightClientSnapshot,
-    update_period = compute_epoch_at_slot(update.header.slot) // EPOCHS_PER_SYNC_COMMITTEE_PERIOD
-    if update_period == snapshot_period + 1:
-        snapshot.current_sync_committee = snapshot.next_sync_committee
-        snapshot.next_sync_committee = update.next_sync_committee
-    snapshot.header = update.header
+def apply_light_client_update(store: LightClientStore, update: LightClientUpdate) -> None:
+    active_header = get_active_header(update)
+    finalized_period = compute_epoch_at_slot(store.finalized_header.slot) // EPOCHS_PER_SYNC_COMMITTEE_PERIOD
+    update_period = compute_epoch_at_slot(active_header.slot) // EPOCHS_PER_SYNC_COMMITTEE_PERIOD
+    if update_period == finalized_period + 1:
+        store.current_sync_committee = store.next_sync_committee
+        store.next_sync_committee = update.next_sync_committee
+    store.finalized_header = active_header
 ```
 
 #### `process_light_client_update`
 
 ```python
-def process_light_client_update(store: LightClientStore, update: LightClientUpdate, current_slot: Slot,
+def process_light_client_update(store: LightClientStore,
+                                update: LightClientUpdate,
+                                current_slot: Slot,
                                 genesis_validators_root: Root) -> None:
-    validate_light_client_update(store.snapshot, update, genesis_validators_root)
-    store.valid_updates.add(update)
-
-    update_timeout = SLOTS_PER_EPOCH * EPOCHS_PER_SYNC_COMMITTEE_PERIOD
+                                
+    validate_light_client_update(store, update, current_slot, genesis_validators_root)
+    
+    # Update the best update in case we have to force-update to it if the timeout elapses
+    if (
+        store.best_valid_update is None
+        or sum(update.sync_committee_bits) > sum(store.best_valid_update.sync_committee_bits)
+    ):
+        store.best_valid_update = update
+    
+    # Track the maximum number of active participants in the committee signatures
+    store.current_max_active_participants = max(
+        store.current_max_active_participants,
+        sum(update.sync_committee_bits),
+    )
+    
+    # Update the optimistic header
+    if (
+        sum(update.sync_committee_bits) > get_safety_threshold(store) and
+        update.attested_header.slot > store.optimistic_header.slot
+    ):
+        store.optimistic_header = update.attested_header
+    
+    # Update finalized header
     if (
         sum(update.sync_committee_bits) * 3 >= len(update.sync_committee_bits) * 2
-        and update.finality_header != BeaconBlockHeader()
+        and update.finalized_header != BeaconBlockHeader()
+    ):
+        # Normal update through 2/3 threshold
+        apply_light_client_update(store, update)
+        store.best_valid_update = None
+    elif (
+        current_slot > store.finalized_header.slot + UPDATE_TIMEOUT
+        and store.best_valid_update is not None
```

Comment:
`store.best_valid_update` cannot be `None` here (in that case, it is initialized with `update` above)